### ISSUE TITLE
Failure when adding new file to project

### ISSUE THREAD

**Issue by eranheres** [wojciech-kulik/xcodebuild.nvim#143]
Doing the following fails:
1. creating a new Swift file
2. running XcodebuildAddCurrentFile

The following error was received:
```
To see more details please check /tmp/xcodebuild_nvimtree.
If you are trying to add files to SPM packages, you may want to filter them out in the config using: integrations.nvim_tree.should_update_project.
If the error is unexpected, please open an issue on GitHub.
```

```
❯ cat /tmp/xcodebuild_nvimtree
<internal:/opt/homebrew/Cellar/ruby/3.3.0/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require': cannot load such file -- xcodeproj (LoadError)
	from <internal:/opt/homebrew/Cellar/ruby/3.3.0/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:127:in `require'
	from /Users/.../.local/share/nvim-ios/lazy/xcodebuild.nvim/tools/project_helper.rb:3:in `<main>'
```

My checkhealth:
```
==============================================================================
xcodebuild: require("xcodebuild.health").check()

Checking xcodebuild.nvim ~
- OK xcodebuild.nvim: commit #3fd5b2e

Checking OS ~
- OK macOS 14+: release 23.3.0

Checking plugins ~
- OK telescope.nvim installed.
- OK nui.nvim installed.
- OK nvim-tree installed.
- WARNING oil.nvim not installed. (Optional to visually manage project files)
- OK nvim-dap installed.
- OK nvim-dap-ui installed.
- OK nvim-treesitter installed.
- OK nvim-treesitter: Swift parser installed.

Checking required dependencies ~
- OK xcodebuild: version 15.3
- OK xcodebuild: installed
- OK xcrun: installed
- OK xcode-build-server: installed

Checking optional dependencies ~
- OK pymobiledevice3: installed
- OK xcodeproj: installed
- OK xcbeautify: installed
- OK ruby: version 3.3.0

Checking debugger ~
- OK nvim-dap: codelldb adapter configured.
- OK nvim-dap: swift configuration found.
- OK codelldb: installed

Checking buildServer.json ~
- OK buildServer.json: found

Checking .nvim/xcodebuild/settings.json ~
- OK .nvim/xcodebuild: found
- OK .nvim/xcodebuild/settings.json: found

Checking passwordless sudo ~
- WARNING sudo: passwordless permission for `remote_debugger` is not configured.
- WARNING debugging on physical devices with iOS 17+ will not work.
- WARNING see `:help |xcodebuild.sudo`| for more information.
```

**Comment by wojciech-kulik** [comment]
Hello,

it looks like something is wrong with your ruby paths (probably). The script doesn't see `xcodeproj` gem.

**Comment by eranheres** [comment]
thanks, I will check it out

**Comment by joeldrotleff** [comment]
I see this issue now too. Not sure when it started as I usually don't add files to project via Neovim, but I remember it used to work before.

**Comment by joeldrotleff** [comment]
Here's my build log:

**Comment by joeldrotleff** [comment]
/Users/joeldrotleff/.local/share/nvim/lazy/xcodebuild.nvim/tools/project_helper.rb:40:in `find_group_by_absolute_dir_path': undefined method `filter_map' for #<Enumerator::Lazy:0x000000015005cae8> (NoMethodError)
Did you mean?  filter
	from /Users/joeldrotleff/.local/share/nvim/lazy/xcodebuild.nvim/tools/project_helper.rb:209:in `find_targets_for_group'
	from /Users/joeldrotleff/.local/share/nvim/lazy/xcodebuild.nvim/tools/project_helper.rb:262:in `add_file'
	from /Users/joeldrotleff/.local/share/nvim/lazy/xcodebuild.nvim/tools/project_helper.rb:287:in `handle_action'
	from /Users/joeldrotleff/.local/share/nvim/lazy/xcodebuild.nvim/tools/project_helper.rb:348:in `<main>'

**Comment by wojciech-kulik** [comment]
@joeldrotleff your issue is caused by outdated ruby version, please update it.

**Comment by joeldrotleff** [comment]
@wojciech-kulik Ok, you're right! Thanks for the help. For reference, what I had was the system Ruby installed (some old version) and a new version installed via Homebrew but it wasn't symlinked (I use fish shell). So I had to do fish_add_path. But that wasn't actually enough, I then did 'gem install xcodeproj' to update it (Xcodeproj was already installed before), and for good measure did 'gem update --system 3.5.10' and now all is good
